### PR TITLE
BUG: Check that values for "nrows" and "chunksize" are valid

### LINKED ISSUE #15767: Proper checks on nrows and chunksize for read_csv
From #15755 (e.g. [this comment](https://github.com/pandas-dev/pandas/pull/15756#issuecomment-288108170)).

Current status:
- there is a check on ``nrows``, which ensure floats are converted to int with no loss of precision, but not that the resulting/passed value is non-negative
- there is no check on ``chunksize``, so that a ``pd.concat`` based on ``chunksize=0`` will enter an infinite loop

I guess we want to ensure ``chunksize > 0`` and ``nrows >= 0``.

### PR DESCRIPTION
**Pull request by toobaz**
- [x] closes #15767
 - [x] tests added / passed
 - [x] passes ``git diff master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -344,24 +344,24 @@
...
+def _validate_integer(name, val, min_val=0):
     """
-    Checks whether the 'nrows' parameter for parsing is either
+    Checks whether the 'name' parameter for parsing is either
     an integer OR float that can SAFELY be cast to an integer
     without losing accuracy. Raises a ValueError if that is
     not the case.
     """
-    msg = "'nrows' must be an integer"
+    msg = "'%s' must be an integer >=%s" % (name, min_val)
```
Comment (jreback) [inline pandas/io/parsers.py]: can you use "{}".format(...) string formatting, ideally with named parameters

File: pandas/io/parsers.py [hunk 2]
```diff
@@ -344,24 +344,24 @@
...
 
-def _validate_nrows(nrows):
+def _validate_integer(name, val, min_val=0):
     """
-    Checks whether the 'nrows' parameter for parsing is either
+    Checks whether the 'name' parameter for parsing is either
     an integer OR float that can SAFELY be cast to an integer
     without losing accuracy. Raises a ValueError if that is
     not the case.
     """
```
Comment (jreback) [inline pandas/io/parsers.py]: can you add a Parameters section to the doc-string.

File: pandas/tests/io/parser/common.py [hunk 3]
```diff
@@ -402,6 +405,18 @@ def test_read_chunksize(self):
         tm.assert_frame_equal(chunks[1], df[2:4])
         tm.assert_frame_equal(chunks[2], df[4:])
 
+        # with invalid chunksize value:
```
Comment (jreback) [inline pandas/tests/io/parser/common.py]: is there a test for nrows that is negative? and same for chunsize?

**Review by jreback — CHANGES_REQUESTED** [review]
some minor comments

**Comment by jorisvandenbossche** [comment]
Looks good!

Currently, `chunksize=0` also 'works' (it was only negatives and floats between 0 and 1 that caused the infinite loop, chunksize=0 is somehow checked for), but just directly returning the full dataframe. I suppose because the 0 was just interpreted as 'False'. 
I don't think we need to keep supporting this, however, as this is rather strange?

File: pandas/tests/io/parser/common.py (hunk 3 above)
Comment (toobaz) [inline pandas/tests/io/parser/common.py]: Well, there is a test for ``nrows=-1`` and one for ``chunksize=0``... which is already "too negative". So yes, there are tests for the positivity check.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. I agree with @jorisvandenbossche point, that ``chunksize==0`` is an error as well.

**Comment by jorisvandenbossche** [comment]
@toobaz Thanks!

**Comment by toobaz** [comment]
You're welcome!

**Pull request merged**